site/src/components/Filter/UserFilter.tsx

```diff
@@ -0,0 +1,100 @@
+import { useMe } from "hooks"
+import { BaseOption } from "./options"
+import { getUsers } from "api/api"
+import { UseFilterMenuOptions, useFilterMenu } from "./menu"
+import { FilterSearchMenu, OptionItem } from "./filter"
+import { UserAvatar } from "components/UserAvatar/UserAvatar"
+
+export type UserOption = BaseOption & {
+  avatarUrl?: string
+}
+
+export const useUserFilterMenu = ({
+  value,
+  onChange,
+  enabled,
+}: Pick<
+  UseFilterMenuOptions<UserOption>,
+  "value" | "onChange" | "enabled"
+>) => {
+  const me = useMe()
+
+  const addMeAsFirstOption = (options: UserOption[]) => {
+    options = options.filter((option) => option.value !== me.username)
+    return [
+      { label: me.username, value: me.username, avatarUrl: me.avatar_url },
+      ...options,
+    ]
+  }
+
+  return useFilterMenu({
+    onChange,
+    enabled,
+    value,
+    id: "owner",
+    getSelectedOption: async () => {
+      const usersRes = await getUsers({ q: value, limit: 1 })
```

Comment:
We call `getUsers` here, and also on [line 48](https://github.com/coder/coder/pull/7878/files#diff-b388ed4298018167d2d62ac103741e6886d048858d09140c5b81ee0b13a45ee3R48). Curious why we can't just make one call to grab all users and then filter on the user we want for `getSelectedOption`.

Reply:
Good question, let's say we have a user called Wesley, Wesley is probably one of the last users to be queried in the database, so when we display the autocomplete options for the users, maybe Wesley is not shown because we only show the first 25 results. So making a separate query to get the selected value is safer and we can be sure we are going to find it.

Reply:
Gotcha, thanks for explaining!